DataProcessing/QuiverWallStreetBetsDataDownloader.cs

```diff
@@ -93,121 +86,56 @@ public bool Run()
-                                        
-                                        if (!_canCreateUniverseFiles)
-                                            continue;
-                                        
-                                        var sid = SecurityIdentifier.GenerateEquity(ticker, Market.USA, true, mapFileProvider, dateTime);
-
-                                        var universeCsvContents = $"{sid},{ticker},{mentions},{rank},{sentiment}";
-
-                                        var queue = _tempData.GetOrAdd(date, new ConcurrentQueue<string>()); 
-                                        queue.Enqueue(universeCsvContents);
-                                    }
-
-                                    if (csvContents.Count != 0)
-                                    {
-                                        SaveContentToFile(_destinationFolder, ticker, csvContents);
-                                    }
-
-                                    var newCompaniesCompleted = Interlocked.Increment(ref companiesCompleted);
-                                    if (newCompaniesCompleted % 100 == 0)
-                                    {
-                                        Log.Trace(
-                                            $"QuiverWallStreetBetsDataDownloader.Run(): {newCompaniesCompleted}/{count} complete");
-                                    }
-                                }
-                            )
-                    );
-
-                    if (tasks.Count == 10)
+                    foreach (var wsbMention in kvp)
                     {
-                        Task.WaitAll(tasks.ToArray());
+                        var ticker = wsbMention.Ticker.ToUpperInvariant();
 
-                        foreach (var kvp in _tempData)
+                        if (!wsbMentionsByTicker.TryGetValue(ticker, out var wsbTickerMentions))
                         {
-                            SaveContentToFile(_universeFolder, kvp.Key, kvp.Value);
+                            wsbMentionsByTicker.Add(ticker, new List<string>());
                         }
 
-                        _tempData.Clear();
-                        tasks.Clear();
+                        wsbMentionsByTicker[ticker].Add($"{date:yyyyMMdd},{wsbMention.Mentions},{wsbMention.Rank},{wsbMention.Sentiment}");
+
+                        if (!_canCreateUniverseFiles) continue;
+
+                        var sid = SecurityIdentifier.GenerateEquity(ticker, Market.USA, true, mapFileProvider, date);
+                        universeCsvContents.Add($"{sid},{ticker},{wsbMention.Mentions},{wsbMention.Rank},{wsbMention.Sentiment}");
                     }
-                }
 
-                if (tasks.Count != 0)
-                {
-                    Task.WaitAll(tasks.ToArray());
-                    
-                    foreach (var kvp in _tempData)
+                    if (_canCreateUniverseFiles && universeCsvContents.Any())
```

Comment:
nit - `universeCsvContents.Any()` is a proxy for `_canCreateUniverseFiles ` really?

Reply:
Yes, they should have the same result in principle so I added the Boolean first and left the second "to be sure".